File: src/main/java/net/superricky/tpaplusplus/util/DeathPos.java
package net.superricky.tpaplusplus.util;

import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.phys.Vec3;
import org.joml.Vector3f;

public class DeathPos extends Vec3 {
    public DeathPos(ServerLevel deathLevel, double pX, double pY, double pZ) {
        super(pX, pY, pZ);
        this.deathLevel = deathLevel;
    }

    public DeathPos(ServerLevel deathLevel, Vector3f pVector) {
        super(pVector);
        this.deathLevel = deathLevel;
    }

    public ServerLevel getDeathLevel() {
        return deathLevel;
    }

    private final ServerLevel deathLevel;
}

File: src/main/kotlin/net/superricky/tpaplusplus/database/DeathPos.kt
package net.superricky.tpaplusplus.database

data class DeathPos(
    var x: Double = 0.0,
    var y: Double = 0.0,
    var z: Double = 0.0,
    var yaw: Float = 0.0f,
    var pitch: Float = 0.0f,
    var world: String = "",
    var backed: Boolean = true
)
